pkg/component/runtime/log_writer.go

```diff
@@ -21,22 +22,47 @@ type zapcoreWriter interface {
 	Write(zapcore.Entry, []zapcore.Field) error
 }
 
+type logSource uint8
+
+const (
+	logSourceStdout logSource = 0
+	logSourceStderr logSource = 1
+)
+
 // logWriter is an `io.Writer` that takes lines and passes them through the logger.
 //
 // `Write` handles parsing lines as either ndjson or plain text.
 type logWriter struct {
-	loggerCore zapcoreWriter
-	logCfg     component.CommandLogSpec
-	remainder  []byte
+	loggerCore   zapcoreWriter
+	logCfg       component.CommandLogSpec
+	logLevel     zap.AtomicLevel
+	unitLevels   map[string]zapcore.Level
+	levelMx      sync.RWMutex
+	inheritLevel zapcore.Level
```

Comment:
What is inheritLevel? It still isn't very clear to me after reading the implementation. Cloud you add a comment?

Reply:
inheritLevel is the level that will be used for a log message in the case it doesn't define a log level for stdout it is INFO and for stderr it is ERROR.

I have added that comment in the code.